### ISSUE TITLE
How do I find the cause of the Jank Slow Frame?

### ISSUE THREAD

**Issue by kevin4dhd** [flutter/devtools#4179]
Hello, I have several frames that take more than 500 ms to complete, I am trying to find out why, in which line of dart code does this slow exist in my app, but it does not tell me in which line this is happening or what is running.

In my widget, I only have one container, nothing more, because the slowness is caused by the dart code, but I would like devtools to tell me what line is making it take too long, this makes it freeze for 2 or 3 seconds, maybe already devtools brings it but I can't find it, I hope you can help me please

![image](https://user-images.githubusercontent.com/22886367/172666630-429b5b4f-2939-497c-ae65-159629ac424d.png)

**Comment by kenzieschmoll** [comment]
In this slow frame, it looks like the Build part of the frame is what is slow. After determining this, one flag you can turn on to help is the Track Widget Builds flag. Click on "Enhance Tracing" above, enable "Track Widget Builds", and then reproduce the janky activity in your app. This will add an additional timeline event to the time for each widget built. 

You'll also see "Track Layouts" and "Track Paints" in the Enhance Tracing menu. These options can be used if you come across a frame where Layout or Paint are the main contributor to time on the UI thread. Please let us know if these options help! thanks.

**Comment by kenzieschmoll** [comment]
Did you reproduce the janky activity in your app once you enabled Track Widget Builds? You should see additional timeline events below the BUILD event: https://docs.flutter.dev/development/tools/devtools/performance#track-widget-builds.

Narrowing down a BUILD event to the Widgets that contributed will help with guiding you to your problem. Another thing you can do is collect a CPU profile to understand what is going on during your slow frame / slow widget builds. You can click the BUILD event here to load a CPU profile for that time range, or you can use the CPU profiler tab to profile your whole app, but you will not be able to see the profile in the context of your slow frame with the latter method.

**Comment by MeshackT** [comment]
Hello Kenzieschmoll.
I tried following the Flutter I/O doc regarding tracking the cause of slow build-up and jank or shaders in my app but I still could not translate my graphs into literal words on what to do.

May you please help me know how to fix slow performance in simpler terms, your help will be highly appreciated.
